### PR TITLE
[Clang] Fix crash on improper use of __array_extent

### LINKED ISSUE #80474: Clang crash on improper use of `__array_extent`
Clang crashes on
```c++
auto test() {
  return __array_extent(int, );
}
```
https://godbolt.org/z/3Wa51enP9

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #80474

### CODE REVIEW

File: clang/test/SemaCXX/incomplete-array-extent.cpp [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// RUN: %clang_cc1 -verify -std=c++11 %s
+
+auto f() { // expected-error {{'auto' return without trailing return type; deduced return types are a C++14 extension}}
```
Comment (Sirraide) [inline clang/test/SemaCXX/incomplete-array-extent.cpp:3]: I’d set the standard version to `C++14` or later to avoid unrelated diagnostics

**Review by Sirraide — APPROVED** [review]
LGTM 

Though I do wonder how many other type traits are also affected by this, because at least the function below this one also doesn’t seem to check for parse errors... it would probably make sense to take a look at the rest of them too, but that can also be a separate pr.

File: clang/test/SemaCXX/incomplete-array-extent.cpp (hunk 1 above)
Comment (a-tarasyuk) [reply clang/test/SemaCXX/incomplete-array-extent.cpp]: Thanks for the review. I've updated test

**Review by cor3ntin — APPROVED** [review]
I've edited the title of the PR (which is then used as commit message)
Please follow that style in the future (see also https://llvm.org/docs/DeveloperPolicy.html#commit-messages)

The change itself looks fine, thanks!

**Comment by a-tarasyuk** [comment]
@cor3ntin Thank you for pointing that document out to me. I initially thought the PR title should match the issue title...

**Pull request merged**